### PR TITLE
Support for tabbed encounters.

### PR DESCRIPTION
**Pull request by sunsetsystems**
Following up on issue #1142, here's my first take on a solution for tabbed encounters. This is ported from an earlier OpenEMR and tested well there, but this version could use some testing assistance.

(below added by bradymiller):
Demo is at : http://www.open-emr.org/wiki/index.php/Development_Demo#Nu_-_Up_For_Grabs_Demo

### CODE REVIEW

File: library/TabsWrapper.class.php [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    function __construct($tabsid = 'tabs')
+    {
+        $this->tabsid = $tabsid;
+    }
+
+    // Declare an initial tab that will not be dynamically created.
+    // Commonly this is called once but you can make as many as you want, or none.
+    //
+    public function declareInitialTab(
+        $title = 'Initial Tab', // Literals not translated here because they are just examples.
```
Comment (sunsetsystems) [inline library/TabsWrapper.class.php]: Travis CI gives an error "Multi-line function declarations must define one parameter per line" here, but I don't see what the problem is.

**Comment by bradymiller** [comment]
Up For Grabs demo for this feature is here: http://www.open-emr.org/wiki/index.php/Development_Demo#Nu_-_Up_For_Grabs_Demo

**Comment by bradymiller** [comment]
Very neat on quick testing! Will take a closer look at the code over next day or 2.

File: interface/patient_file/encounter/encounter_top.php [hunk 2]
```diff
@@ -1,12 +1,33 @@
...
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see
+ * http://www.gnu.org/licenses/licenses.html#GPL .
+ *
+ * @package OpenEMR
+ * @license http://www.gnu.org/licenses/licenses.html#GPL GNU GPL V3+
+ * @author  Rod Roark <rod@sunsetsystems.com>
+ * @link    http://www.open-emr.org
+ */
```
Comment (bradymiller) [inline interface/patient_file/encounter/encounter_top.php]: Rec using the newer header/copyright/author template:
http://www.open-emr.org/wiki/index.php/How_to_Document_Your_Code_Properly#File-Level_DocBlock

File: library/TabsWrapper.class.php [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see
+ * http://www.gnu.org/licenses/licenses.html#GPL .
+ *
+ * @package OpenEMR
+ * @license http://www.gnu.org/licenses/licenses.html#GPL GNU GPL V3+
+ * @author  Rod Roark <rod@sunsetsystems.com>
+ * @link    http://www.open-emr.org
+ */
```
Comment (bradymiller) [inline library/TabsWrapper.class.php]: Rec using the newer header/copyright/author template:
http://www.open-emr.org/wiki/index.php/How_to_Document_Your_Code_Properly#File-Level_DocBlock

File: library/TabsWrapper.class.php [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+
+class TabsWrapper
```
Comment (bradymiller) [inline library/TabsWrapper.class.php]: This class should be in the autoloader. Let me know if need guidance on how to do this. And would place this in the library/tabs/src/ directory.

**Comment by bradymiller** [comment]
Should also update the xmlformgen and formmaker code with the new close code.(this will be straightforward; just let me know if need guidance)

**Comment by bradymiller** [comment]
review complete. :)

File: library/TabsWrapper.class.php (hunk 4 above)
Comment (bradymiller) [reply library/TabsWrapper.class.php]: Also would place it in the `OpenEMR\Tabs` namespace

**Comment by sunsetsystems** [comment]
Thanks Brady, I should be able to get back to this over the weekend.

**Comment by sunsetsystems** [comment]
Suggested changes have been applied.

**Comment by sunsetsystems** [comment]
By the way I'm wondering what the namespace buys us, given that TabsWrapper is the only name in it. :)

**Comment by sunsetsystems** [comment]
@bradymiller Did you want to look at this again, or should I proceed to merge?

**Comment by bradymiller** [comment]
Looks good. The only thing to ensure PSR4 for the class is to remove the .class from the class filename. Then I think it's ready for the codebase,

**Comment by bradymiller** [comment]
Also, regarding the TabsWrapper namespace, basically is to provide a nice and warm fuzzy feeling :)
The nice thing about these paths/namespaces is that they are super easy to move around and reorganize, when needed.

**Pull request merged**